simulation/test_swarm.cpp

```diff
@@ -558,17 +558,14 @@ TORRENT_TEST(block_uploaded_alert)
 			if (auto at = lt::alert_cast<lt::add_torrent_alert>(a))
 			{
 				// init blocks vector, MUST happen before any block_uploaded_alert alerts
-				std::shared_ptr<lt::torrent> torrent = at->handle.native_handle();
-				std::shared_ptr<const torrent_info> torrent_info = torrent->get_torrent_copy();
+				auto torrent = at->handle.native_handle();
+				auto torrent_info = &torrent->torrent_file();
```

Comment:
Ok so remove `auto torrent_info = &torrent->torrent_file();` and just call `at->handle.torrent_file()` the 2 times `torrent_info` is used:

```c++
...
auto torrent_info = &torrent->torrent_file();
int blocks_per_piece = torrent_info->piece_length() / torrent->block_size();
blocks.resize(torrent_info->num_pieces(), std::vector<bool>(blocks_per_piece, false));
...
```

Becomes:

```c++
...
int blocks_per_piece = at->handle.torrent_file()->piece_length() / torrent->block_size();
blocks.resize(at->handle.torrent_file()->num_pieces(), std::vector<bool>(blocks_per_piece, false));
...
```


?